pkg/dataset/types.go

```diff
@@ -53,13 +56,24 @@ func (rM *RemediationIdsMap) ContainsID(r remediation.Remediation, id int64) (in
 }
 
 func (rM *RemediationIdsMap) AddID(clog *log.Entry, r remediation.Remediation, id int64, origin string) {
+	// Initialize map if nil
+	if *rM == nil {
+		*rM = make(RemediationIdsMap, 1) // Pre-allocate for 1 remediation type
+	}
+
 	ids, ok := (*rM)[r]
 	if !ok {
-		clog.Tracef("remediation %s not found, creating", r.String())
-		(*rM)[r] = []RemediationDetails{{id, origin}}
+		if clog != nil && clog.Logger.IsLevelEnabled(log.TraceLevel) {
+			clog.Tracef("remediation %s not found, creating", r.String())
+		}
+		// Pre-allocate slice with capacity for multiple IDs per remediation
+		(*rM)[r] = make([]RemediationDetails, 0, 4) // Start with capacity 4
```

Comment:
questionable, might be better to start from one 🤷🏻 as we expect 1 decision really per remediation